### PR TITLE
[SCFToCalyx] Add conversion for DivUIOP and RemUIOp. 

### PR DESCRIPTION
**Pull request by DeepFlyingSky**
This PR deals with [issue-2141](https://github.com/llvm/circt/issues/2141).
Mainly follow the code in Add conversion for MulIOp.
But donot find any modulo operation defined in the 'SCF' or 'arith' Dialect. Is there any missing?

### CODE REVIEW

**Comment by mikeurbach** [comment]
I believe you will find [remui](https://mlir.llvm.org/docs/Dialects/ArithmeticOps/#arithremui-mlirarithremuiop) in the arith dialect for unsigned integer modulo operation.

**Review by cgyurgyik — COMMENTED** [review]
Thanks!!

Here is the unsigned modulo operator: https://mlir.llvm.org/docs/Dialects/ArithmeticOps/#arithremui-mlirarithremuiop

Since DivUIOp, RemUIOp, and MulIOp only differ in a few places (output, register name, types), it _might_ be possible to create a helper function to reduce a lot of the code duplication. However, the number of differences might prove this effort futile.

**Comment by dodohack** [comment]
> Since DivUIOp, RemUIOp, and MulIOp only differ in a few places (output, register name, types), it _might_ be possible to create a helper function to reduce a lot of the code duplication. However, the number of differences might prove this effort futile. WDYT?

Agree, I think a template function similar to `buildLibraryOp` should created for Div/Rem/Mul ops.

**Comment by DeepFlyingSky** [comment]
Thank @mikeurbach , @cgyurgyik , @dodohack . 
Add the RemUIOp conversion.
Change the code in DivUIOp, RemUIOp, and MulIOp  with a template function.

File: lib/Conversion/SCFToCalyx/SCFToCalyx.cpp
Comment (mikeurbach) [inline lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:915]: nitpick: I think this comment should be a sentence, like `build a TCalyxLibPipeOp, to deal with ...`

**Review by mikeurbach — APPROVED** [review]
Thanks for the revisions, this is looking good. I had one minor style nitpick. The meat of this looks great. I've approved CI to run. Let us know if you need one of us to merge this. You can request write access to CIRCT so you can merge your own PRs according to the MLIR/LLVM policy here: https://mlir.llvm.org/getting_started/Contributing/#first-time-contributors

File: lib/Conversion/SCFToCalyx/SCFToCalyx.cpp
Comment (cgyurgyik) [inline lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:918]: Nit: Can we also specify in the name this is for binary pipeline operations?

File: test/Conversion/SCFToCalyx/convert_simple.mlir
Comment (cgyurgyik) [inline test/Conversion/SCFToCalyx/convert_simple.mlir:127]: Nit: add extra newline please.

File: lib/Conversion/SCFToCalyx/SCFToCalyx.cpp
Comment (cgyurgyik) [inline lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:1048]: Nit: Can we either give this a type, or just use `mulPipe.out()` below, since it is only used once? You could even do
```cpp
buildLibraryPipeOp<calyx::MultPipeLibOp>(rewriter, mul, mulPipe, /*out=*/mulPipe.out());
```

File: lib/Conversion/SCFToCalyx/SCFToCalyx.cpp
Comment (cgyurgyik) [inline lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:1060]: Same here.

File: lib/Conversion/SCFToCalyx/SCFToCalyx.cpp
Comment (cgyurgyik) [inline lib/Conversion/SCFToCalyx/SCFToCalyx.cpp:1072]: Same here.

**Review by cgyurgyik — APPROVED** [review]
Thank you for this! This bug has been open for awhile now and the help is very much appreciated. Approved with a few nit picks.

**Comment by DeepFlyingSky** [comment]
Thank @mikeurbach , @cgyurgyik .
Nit all fix.
And I send a email to request the commit access.

**Comment by mikeurbach** [comment]
Thanks @DeepFlyingSky. I just enabled the CI to run on the latest revision, and will merge once it passes.

**Comment by mikeurbach** [comment]
Oops, looks like clang-format had an issue: https://github.com/llvm/circt/runs/5203614522?check_suite_focus=true#step:9:1. I believe we use the latest clang-format from https://releases.llvm.org/download.html in CI.

**Comment by DeepFlyingSky** [comment]
Thanks for your help @mikeurbach . 
Format all fixes now.

**Comment by mikeurbach** [comment]
Awesome, merged. Thanks for the contribution.

**Pull request merged**
